src/app.ts

```diff
@@ -110,6 +175,25 @@ export function buildApp(
 
 	app.use(jsonBodyParser());
 
+	const uiHandler = (req: express.Request, res: express.Response) => {
+		withPandaAuth(
+			getPanda,
+			req,
+			res,
+			() => {
+				res.contentType('html').send(getUI());
+			},
+			() => {
+				res.status(403)
+					.contentType('html')
+					.send(getRedirectResponse(req));
```

Comment:
I reckon it'd be fine to immediately redirect to the login service here, rather than return a 403 and use `http-equiv="refresh"` to do a client side redirect. Or alternatively we could return HTML with a login link. I think both those patterns are used by other tools hosted on gutools.co.uk domains. What do you think?